src/server/worker/pipeline/transform/chain/chain.go

```diff
@@ -130,7 +132,7 @@ func (jdi *JobDatumIterator) Iterate(cb func(*datum.Meta) error) error {
 			if outputFilesetID, err = jdi.withDatumFileset(pachClient, func(outputSet *datum.Set) error {
 				return datum.Merge([]datum.Iterator{parentFilesetIterator, filesetIterator}, func(metas []*datum.Meta) error {
 					if len(metas) == 1 {
-						if metas[0].PipelineJobID != jdi.pipelineJobID {
+						if !proto.Equal(metas[0].PipelineJob, jdi.pipelineJob) {
```

Comment:
will this ever actually be considering datum metadata from another pipeline?
happy with comparing the protos regardless

Reply:
no, the chain is only run in the worker, which is for a single pipeline